### DISCUSSION TITLE
Length of time to activate an effect

### DISCUSSION THREAD

**Discussion by DanDixon** [Anders-Holst/xled_plus#16] [Q&A] [answered]
Even with setting a manual IP address of the twinkly lights, it takes about 7-8 seconds from when xled_plus is started until the lights change. Is this expected?

How long does it take from starting xled_plus until the lights change for others?

Is there anything that can be done to make the delay shorter?

Is there any way to keep the communication path open instead of reconnecting each time? (or would that require a significant change to how xled is structured?)

**Comment by Anders-Holst** [comment, marked as answer]
The time depends on what you try to do. If I just turn them on or off, or switch between already uploaded movies, then it is fast. However, if you create and upload a new movie, as with most of the examples published here, the uploading takes several seconds. This is the same if you upload a longer movie from the app - it also takes some time. Some of the effects defined in here though are very long movies, which may take more than a minute to load.
